src/context.rs

```diff
@@ -236,7 +237,48 @@ impl Context {
     pub fn emit_event(&self, event: EventType) {
         self.events.emit(Event {
             id: self.id,
-            typ: event,
+            typ: event.clone(),
+        });
+        let context = self.clone();
+        async_std::task::spawn(async move {
+            // TODO synchronously is prob. better
+            match context.get_config_int(Config::DebugLogging).await {
```

Comment:
`get_config_int` is slow even if cached, because internally it calls `get_raw_config` which accepts `AsRef<str>` as a key. So `Config::DebugLogging` is converted to string and the string is hashed just to lookup the cache.

Cache should use `Config` enum as a key instead.

Reply:
Turns out that what was slow is the `task::spawn()`, not `get_config_int()`. And using `Config` enum as the key would have the disadvantage that it wouldn't cache some settings that are set by string using `set_raw_config()` (`ui.*` configs among them).

Reply:
> 

gonna fix that later too

Reply:
> gonna fix that later too

Now I'm confused... The slowness is resolved by using the context.debug_logging AtomicU32. We can still switch to using `block_on`, but discussion for this is at https://github.com/deltachat/deltachat-core-rust/pull/3296#discussion_r866367352